wipac_dev_tools/logging_tools.py

```diff
@@ -66,18 +66,32 @@ def log_argparse_args(
 
 
 def log_dataclass(
-    dclass: T, logger: Union[str, logging.Logger], level: LoggerLevel
+    dclass: T,
+    logger: Union[str, logging.Logger],
+    level: LoggerLevel,
+    prefix: str = "",
+    obfuscate_substrings: Optional[List[str]] = None,
 ) -> T:
-    """Log a dataclass instance's fields and members."""
+    """Log a dataclass instance's fields and members.
+
+    `obfuscate_substrings` is case-insensitive
+    """
     import dataclasses  # imports for python 3.7+
 
     if not (dataclasses.is_dataclass(dclass) and not isinstance(dclass, type)):
         raise TypeError(f"Expected instantiated dataclass: 'dclass' ({dclass})")
 
     logger_fn = get_logger_fn(logger, level)
 
+    if not obfuscate_substrings:
+        obfuscate_substrings = []
+    obfuscate_substrings = [o.upper() for o in obfuscate_substrings]
+
     for field in dataclasses.fields(dclass):
-        logger_fn(f"(env) {field.name}: {getattr(dclass, field.name)}")
+        val = getattr(dclass, field.name)
+        if any(s.upper() in obfuscate_substrings for s in field.name):
```

Comment:
If I'm reading this right, it's testing individual chars in a field name?  Maybe instead you want
```suggestion
        if any(s in field.name.upper() for s in obfuscate_substrings):
```